Floofbot/Modules/TagCommands.cs

```diff
@@ -0,0 +1,206 @@
+
+        public TagCommands(FloofDataContext floofDb)
+        {
+            _floofDb = floofDb;
+        }
+
+        [Command("add")]
+        [Priority(0)]
+        [RequireUserPermission(GuildPermission.AttachFiles)]
+        public async Task Add(
+            [Summary("Tag name")] string tag,
+            [Summary("Tag content")] [Remainder] string content = null)
+        {
+            if (content != null)
+            {
+                Regex rgx = new Regex("[^a-zA-Z0-9 -]");
+                tag = rgx.Replace(tag, "").ToLower();
+                string tagName = $"{tag.ToString()}:{Context.Guild.Id}";
+
+                try
+                {
+                    _floofDb.Add(new Tag
+                    {
+                        TagName = tagName,
+                        ServerId = Context.Guild.Id,
+                        UserId = Context.User.Id,
+                        TagContent = content
+                    });
+                    _floofDb.SaveChanges();
+                    await SendEmbed(CreateDescriptionEmbed($"💾 Added Tag `{tag}`"));
+                }
+                catch (DbUpdateException e)
+                {
+                    await SendEmbed(CreateDescriptionEmbed($"💾 Tag `{tag}` Already Exists"));
+                    Console.WriteLine(e);
+                }
+            }
+            else
+            {
+                await SendEmbed(CreateDescriptionEmbed($"💾 Usage: `tag add [name] [content]`"));
+            }
+        }
+
+        [Command("add")]
+        [Priority(1)]
+        public async Task Add()
+        {
+            await SendEmbed(CreateDescriptionEmbed($"💾 Usage: `tag add [name] [content]`"));
+        }
+
+        [Command("list")]
+        [Summary("Lists all tags")]
+        public async Task ListTags([Remainder] string content = null)
+        {
+            List<Tag> tags = _floofDb.Tags.AsQueryable()
+                .Where(x => x.ServerId == Context.Guild.Id)
+                .OrderBy(x => x.TagName)
+                .ToList();
+            List<string> pages = new List<string>();
+
+            int index = 0;
+            for (int i = 1; i <= (tags.Count / 50) + 1; i++)
+            {
+                string text = "```glsl\n";
+                int pagebreak = index;
+                for (; index < pagebreak + 50; index++)
+                {
+                    if (index < tags.Count)
+                    {
+                        text += $"[{index}] - {tags[index].TagName}\n";
+                    }
+                }
+
+                text += "\n```";
+                pages.Add(text);
+            };
+
+            // await PagedReplyAsync(pages);
+        }
+
+        [Command("remove")]
+        [Summary("Removes a tag")]
+        [RequireUserPermission(GuildPermission.ManageMessages)]
+        public async Task Remove([Summary("Tag name")] string tag)
+        {
+            string tagName = $"{tag.ToLower()}:{Context.Guild.Id}";
```

Comment:
do we still need to use guild id in the tag name if we are adding guild id to the db?

Reply:
It honestly depends on how we'll migrate the data over from SQLite to MySQL.
If we can manually adjust the data before moving it over, that's nice.
This current approach is assuming that tagName is being used for everything right now, though.

Reply:
can be updated in a future pr i guess